### PR TITLE
Critical flows: Shopper → Checkout → Can view empty checkout message

### LINKED ISSUE #5770: Critical flows: Shopper → Checkout → Can view empty checkout message
### Story

As a shopper, I want to be redirected to the cart page when go to the checkout page, but my cart is empty.

### File

`tests/e2e/specs/shopper/checkout-redirect.test.js`

### Test

1. Go to the checkout page.
2. Verify that the empty checkout message is visible.

### PR DESCRIPTION
**Pull request by tarunvijwani**
Fixes #5770

This PR adds e2e tests to ensure that shoppers can see an empty cart message on the checkout page when there are no products in their cart.

#### Automated Tests
* [x] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [x] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Check out this PR.
2. Start Docker Desktop.
3. Run `npm run wp-env destroy && npm run wp-env start` to start a fresh e2e environment.
4. Run `npm run test:e2e-dev -- tests/e2e/specs/shopper/checkout-empty.test.js` to run only this test case.
5. Make sure all tests pass.

### CODE REVIEW

File: tests/e2e/specs/shopper/checkout-redirect.test.js [hunk 1]
```diff
@@ -2,3 +2,27 @@
  * Internal dependencies
  */
 import { shopper } from '../../../utils';
+import { BASE_URL } from '../../utils';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
```
Comment (nielslange) [inline tests/e2e/specs/shopper/checkout-redirect.test.js]: `block.name` seems undefined here. Given that this part only gets called for code that is not hidden behind a flag, I'd change that part from `skipping ${ block.name } tests` to `Skipping checkout tests` or to `Skipping redirection tests`.

In general, I'd question this part, as I believe that we should, at least, also test functionality that behind the featured flag. But that's a different story.

File: tests/e2e/specs/shopper/checkout-redirect.test.js [hunk 2]
```diff
@@ -2,3 +2,27 @@
...
+	beforeAll( async () => {
+		await shopper.block.emptyCart();
+	} );
+
+	it( 'Can see redirect to empty Cart', async () => {
+		const slug = 'checkout';
+		const url = BASE_URL + slug;
+		await page.goto( url, {
+			waitUntil: 'networkidle0',
+		} );
```
Comment (nielslange) [inline tests/e2e/specs/shopper/checkout-redirect.test.js]: Why not using `await shopper.block.goToCheckout()` here? It does the same trick, but is less code. After all, we're only testing if calling the empty checkout page redirects to the cart page and shows the corresponding message then.

**Review by nielslange — APPROVED** [review]
Pre-approving this PR, as in general, the PR is in a good shape. To turn it from a good into a great PR, I left some feedback for you. 😀

**Comment by tarunvijwani** [comment]
@nielslange I have pushed the new test case which we discussed earlier.  Thank you!

File: tests/e2e/specs/shopper/checkout-empty.test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/**
+ * Internal dependencies
+ */
+import { shopper } from '../../../utils';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
```
Comment (nielslange) [inline tests/e2e/specs/shopper/checkout-empty.test.js]: The PR looks great! The only thing I'd still change would be this part:

```suggestion
	test.only( `Skipping checkout tests`, () => {} );
```

**Review by nielslange — APPROVED** [review]
I just left one more comment. I'd change the part I've pointed out, then I'd say we're good to merge. 😀

**Pull request merged**